File: tests/compile-fail/ForeignNullableGenericConcretePropertySlot.cs
using System;
using System.Collections.Generic;

// A CONCRETE virtual property whose type crosses. Unlike the abstract twin, nothing obliges a deriving type to fill
// this one — so the refusal has to reach it through "does this type actually override it", and for a getter that
// question is decided on a ZERO-parameter signature. The sibling property is the control: overriding it is an
// ordinary program and is driven in tests/interop.
namespace plainnet
{
    public class PropBase
    {
        public virtual List<int?> Items { get { return new List<int?>(); } }

        public virtual string Tag { get { return "net"; } }
    }
}

File: tests/compile-fail/ForeignNullableGenericConcretePropertySlot.kt
// #86 — a CONCRETE virtual property whose type crosses, overridden.
//
// The abstract accessor case is refused because an instantiable type must fill the slot at all; this one is only
// this type's problem because the author OVERRODE it, which is the arm that asks what the override physically
// states. A getter has no parameters, so that question is answered on the member's name and generic arity alone —
// which is the whole of a getter's CLR identity.
import plainnet.PropBase
import System.Collections.Generic.List

class CConcreteProp : PropBase() {
    override val Items: List<Int?> get() = List<Int?>()
}

fun main() {
    println(CConcreteProp().Tag.substring(0, 2))
}
